packs/data/feats.db/viking-shieldbearer.json

```
@@ -19,14 +19,51 @@
         "level": {
             "value": 1
         },
-        "location": "",
+        "location": null,
         "prerequisites": {
             "value": []
         },
         "rules": [
             {
                 "key": "GrantItem",
                 "uuid": "Compendium.pf2e.feats-srd.Shield Block"
+            },
+            {
+                "choices": [
+                    {
+                        "label": "PF2E.Weapon.Base.longsword",
+                        "value": "longsword"
+                    },
+                    {
+                        "label": "PF2E.Weapon.Base.battle-axe",
+                        "value": "battle-axe"
+                    }
+                ],
+                "key": "ChoiceSet",
+                "prompt": "PF2E.SpecificRule.Human.VikingShieldbearer.Prompt",
+                "rollOption": "viking"
+            },
+            {
+                "key": "ActiveEffectLike",
+                "mode": "upgrade",
+                "path": "data.martial.weapon-base-longsword.rank",
+                "predicate": {
+                    "all": [
+                        "viking:longsword"
+                    ]
+                },
+                "value": 1
+            },
+            {
+                "key": "ActiveEffectLike",
+                "mode": "upgrade",
+                "path": "data.martial.weapon-base-battle-axe.rank",
+                "predicate": {
+                    "all": [
+                        "viking:battle-axe"
+                    ]
+                },
+                "value": 1
             }
```

```suggestion
            {
                "key": "ActiveEffectLike",
                "mode": "upgrade",
                "path": "data.martial.weapon-base-{item|flags.pf2e.rulesSelections.weaponChoice}.rank",
                "value": 1
            }
```
Path can just be resolved directly and we can add a flag to give it a nice name to refer to instead of spelling out the feat name.